src/foundry/client/pixi/webgl/shaders/lighting/base-lighting.d.mts

```diff
@@ -0,0 +1,182 @@
+
+    static override CONSTANTS: string;
+
+    static override vertexShader: string;
+
+    /**
+     * Construct adaptive shader according to shader type
+     * @param shaderType - shader type to construct : coloration, illumination, background, etc.
+     * @returns the constructed shader adaptive block
+     */
+    static getShaderTechniques(shaderType: AdaptiveLightingShader.ShaderTypes): string;
+
+    /**
+     * The coloration technique coloration shader fragment
+     */
+    static get COLORATION_TECHNIQUES(): ReturnType<(typeof AdaptiveLightingShader)["getShaderTechniques"]>;
+
+    /**
+     * The coloration technique illumination shader fragment
+     */
+    static get ILLUMINATION_TECHNIQUES(): ReturnType<(typeof AdaptiveLightingShader)["getShaderTechniques"]>;
+
+    /**
+     * The coloration technique background shader fragment
+     */
+    static get BACKGROUND_TECHNIQUES(): ReturnType<(typeof AdaptiveLightingShader)["getShaderTechniques"]>;
+
+    /**
+     * The adjustments made into fragment shaders
+     */
+    static get ADJUSTMENTS(): string;
+
+    /**
+     * Contrast adjustment
+     */
+    static CONTRAST: string;
+
+    /**
+     * Saturation adjustment
+     */
+    static SATURATION: string;
+
+    /**
+     * Exposure adjustment
+     */
+    static EXPOSURE: string;
+
+    /**
+     * Switch between an inner and outer color, by comparing distance from center to ratio
+     * Apply a strong gradient between the two areas if attenuation uniform is set to true
+     */
+    static SWITCH_COLOR: string;
+
+    /**
+     * Shadow adjustment
+     */
+    static SHADOW: string;
+
+    /**
+     * Transition between bright and dim colors, if requested
+     */
+    static TRANSITION: string;
+
+    /**
+     * Incorporate falloff if a attenuation uniform is requested
+     */
+    static FALLOFF: string;
+
+    /**
+     * Compute illumination uniforms
+     */
+    static COMPUTE_ILLUMINATION: string;
+
+    /**
+     * Initialize fragment with common properties
+     */
+    static FRAGMENT_BEGIN: string;
+
+    /**
+     * Shader final
+     */
+    static FRAGMENT_END: string;
+
+    /**
+     * A mapping of available coloration techniques
+     */
+    static SHADER_TECHNIQUES: AdaptiveLightingShader.ShaderTechniques;
+    /**
+     * Determine the correct penalty to apply for a given darkness level and luminosity
+     * @param darknessLevel - The current darkness level on [0,1]
+     * @param luminosity    - The light source luminosity on [-1,1]
+     * @returns 0
+     * @deprecated since v12, until v14
+     * @remarks AdaptiveLightingShader#getDarknessPenalty is deprecated without replacement.
+     * The darkness penalty is no longer applied on light and vision sources.
+     */
+    getDarknessPenalty(darknessLevel: number, luminosity: number): 0;
+  }
+
+  namespace AdaptiveLightingShader {
+    type ShaderTypes = "coloration" | "illumination" | "background" | string;
```

Comment:
For intellisense reasons please write `"coloration" | "illumination" | "background" | (string & {})`

TypeScript aggressively simplifies unions so currently this will show in intellisense as simply `string`. However it recognizes the special pattern of `(T & {})` because several libraries were doing it to get better intellisense before TypeScript got more aggressive about simplifying unions with intersections. So now it special cases it for documentation reasons.